File: go/contains-duplicate-217.go
/*
Author: Resul Emre AYGAN
*/

package _go

func containsDuplicate(nums []int) bool {
	if len(nums) <= 1 {
		return false
	}

	resultSet := make(map[int]struct{})

	for _, num := range nums {
		if _, isExists := resultSet[num]; isExists {
			return true
		}

		resultSet[num] = struct{}{}
	}

	return false
}

File: python/contains_duplicate_217.py
"""
Author: Resul Emre AYGAN
"""

from typing import List


class Solution:
    @staticmethod
    def contains_duplicate(nums: List[int]) -> bool:
        number_list = set()

        for number in nums:
            if number in number_list:
                return True

            number_list.add(number)
        return False
